### ISSUE TITLE
static func declarations are skipped, and their body is added to the function above

### ISSUE THREAD

**Issue by rganz** [graydwarf/godot-gdscript-linter#17]
`_analyze_functions` in `analyzer/checkers/function-checker.gd` opens a function only on `trimmed.begins_with("func ")` (line 26). A `static func` declaration does not match, so two things happen at once.

The static function is never analysed. None of the length, parameter, nesting, complexity or return type checks ever see it.

Its body is absorbed into whatever function precedes it, because the parser is still inside that one. The preceding function's reported length and complexity include the static function's body.

## Repro

```gdscript
class_name Probe
extends Node

func small() -> void:
    print("one")

static func large() -> void:
    var a := 1
    var b := 2
    var c := 3
    # ... any number of further lines
```

`small` is reported with the combined length of both bodies. `large` is reported not at all.

## Why it is easy to miss

Both symptoms point away from the real cause. The inflated function looks like a genuine finding, so it gets refactored rather than investigated, and the missing one produces no output to notice. A project can carry this for months and read it as its own code being long.

## Suggested fix

Normalising the line before the `begins_with("func ")` test is enough, for example stripping a leading `static ` so that the existing offset assumptions in `_parse_function_signature` still hold.

This is the same root as #15: the parser decides what a declaration is from the text of one line, so anything in front of `func` breaks it. An annotation breaks it in #15 and `static` breaks it here. A fix that reads the declaration rather than the line prefix would close both.

Measured against the current `main`, plugin version 3.3.0.
